fuel-tx/src/transaction/metadata.rs

```diff
@@ -84,7 +84,7 @@ impl CommonMetadata {
             .iter()
             .map(|input| {
                 let i = offset;
-                offset += input.size();
+                offset = offset.saturating_add(input.size());
```

Comment:
Added validation in 042db352. Might still be a breaking change, since I created new `ValidityError` variants for these. The errors didn't seem to be a good match with this.